cni/cmd/istio-cni-repair/main.go

```diff
@@ -204,3 +225,32 @@ func main() {
 		}
 	}
 }
+
+func setupMonitoring(addr, path string, stop chan struct{}) {
+	mux := http.NewServeMux()
+	var listener net.Listener
+	var err error
+	if listener, err = net.Listen("tcp", addr); err != nil {
+		log.Errorf("unable to listen on socket: %v", err)
+	}
+	exporter, err := ocprom.NewExporter(ocprom.Options{Registry: prometheus.DefaultRegisterer.(*prometheus.Registry)})
+	if err != nil {
+		log.Errorf("could not set up prometheus exporter: %v", err)
+	}
+	view.RegisterExporter(exporter)
+	mux.Handle(path, exporter)
+	monitoringServer := &http.Server{
+		Handler: mux,
+	}
+	go func() {
+		err = monitoringServer.Serve(listener)
+		if err != nil {
+			log.Errorf("could not start monitoring http server: %s", err)
```

Comment:
This message could be confusing? I think it not only returns error at monitoring start?